sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Simsensor reader reads and parses data form special scenario file
+ * than it pushes it to event_queue
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+
+#include "simsensor_reader.h"
+
+#define READER_END_SCENARIO_INDICATOR NO_SENSOR
+#define READER_DATA_SEPARATOR         ','
+
+#define READER_TIMESTAMP_NOT_SET -1
+
+#define READER_ACCEL_DEV_ID 1
+#define READER_BARO_DEV_ID  2
+
+
+typedef struct {
+	time_t first_scenario_time;
+	time_t last_scenario_timestamp;
+	int loops_cnt;
+} simsens_time_t;
+
+
+struct simsens_reader_t {
+	FILE *scenario_file;
+	int sensor_types;
+
+	time_t time_horizon;
+
+	simsens_time_t time_struct;
+};
+
+
+static void reader_timeInit(simsens_time_t *timeStruct)
+{
+	timeStruct->first_scenario_time = READER_TIMESTAMP_NOT_SET;
+	timeStruct->last_scenario_timestamp = READER_TIMESTAMP_NOT_SET;
+	timeStruct->loops_cnt = 0;
+}
+
+
+simsens_reader_t *reader_open(const char *path, int sensor_types, time_t time_horizon)
+{
+	simsens_reader_t *result = malloc(sizeof(simsens_reader_t));
+	if (result == NULL) {
+		return NULL;
+	}
+
+	result->scenario_file = fopen(path, "r");
+	if (result->scenario_file == NULL) {
+		free(result);
+		return NULL;
+	}
+
+	result->sensor_types = sensor_types;
+	result->time_horizon = time_horizon;
+
+	reader_timeInit(&result->time_struct);
+
+	return result;
+}
+
+
+void reader_close(simsens_reader_t *reader)
+{
+	if (reader == NULL) {
+		return;
+	}
+
+	fclose(reader->scenario_file);
+	free(reader);
+}
+
+
+static inline int reader_isLineHeader(const char *line)
+{
+	return line[0] == 's' || line[0] == 'S';
+}
+
+
+static long int reader_getFieldLong(const char *actField, char **nextField, int *error)
```

Comment:
More typical approach: let`s return the error and use pointer argument as the place to store the read value. This way you will evade situations like one couple lines below where you both: set error to -1 and return -1:
```suggestion
static int reader_getFieldLong(const char *fld, char **nextFld, long int *res)
```
btw - `long int` in our system is the length of `int`. How about using specific value length types from `stdint`: `int32`/`int64`/`uint64`? These are the storage type specifiers in `libsensors.h`.

Reply:
Ok, but in this approach I have to define separate function for every type, which have to be parsed

Reply:
But, am also using it for parsing other fields, especially these specific for sensor event and there is many different like `int16`, `uint16`, `uint32`, `int32`, `int64` and so on

Reply:
So `int64` should be wide enough to cover all mentioned cases. The problem is only with sufficiently big `time_t` which is `unsigned long long`.